packages/typespec-client-generator-core/src/example.ts

```diff
@@ -0,0 +1,631 @@
+ */
+async function loadExamples<TServiceOperation extends SdkServiceOperation>(
+  context: TCGCContext,
+  client: SdkClientType<TServiceOperation>
+): Promise<[Map<string, Record<string, LoadedExample>>, readonly Diagnostic[]]> {
+  const diagnostics = createDiagnosticCollector();
+  if (!context.examplesDirectory) {
+    return diagnostics.wrap(new Map());
+  }
+
+  const apiVersion = getValidApiVersion(context, client.apiVersions);
+
+  const exampleDir = apiVersion
+    ? resolvePath(context.examplesDirectory, apiVersion)
+    : resolvePath(context.examplesDirectory);
+  try {
+    if (!(await context.program.host.stat(exampleDir)).isDirectory())
+      return diagnostics.wrap(new Map());
+  } catch (err) {
+    diagnostics.add(
+      createDiagnostic({
+        code: "example-loading",
+        messageId: "noDirectory",
+        format: { directory: exampleDir },
+        target: NoTarget,
+      })
+    );
+    return diagnostics.wrap(new Map());
+  }
+
+  const map = new Map<string, Record<string, LoadedExample>>();
+  const exampleFiles = await context.program.host.readDir(exampleDir);
+  for (const fileName of exampleFiles) {
+    try {
+      const exampleFile = await context.program.host.readFile(resolvePath(exampleDir, fileName));
+      const example = JSON.parse(exampleFile.text);
+      if (!example.operationId || !example.title) {
+        diagnostics.add(
+          createDiagnostic({
+            code: "example-loading",
+            messageId: "noOperationId",
+            format: { filename: fileName },
+            target: NoTarget,
+          })
+        );
+        continue;
+      }
+
+      if (!map.has(example.operationId)) {
+        map.set(example.operationId.toLowerCase(), {});
+      }
+      const examples = map.get(example.operationId.toLowerCase())!;
+
+      if (example.title in examples) {
+        diagnostics.add(
+          createDiagnostic({
+            code: "duplicate-example-file",
+            target: NoTarget,
+            format: {
+              filename: fileName,
+              operationId: example.operationId,
+              title: example.title,
+            },
+          })
+        );
+      }
+
+      examples[example.title] = {
+        relativePath: fileName,
+        file: exampleFile,
+        data: example,
+      };
+    } catch (err) {
+      diagnostics.add(
+        createDiagnostic({
+          code: "example-loading",
+          messageId: "default",
+          format: { filename: fileName, error: err?.toString() ?? "" },
+          target: NoTarget,
+        })
+      );
+    }
+  }
+  return diagnostics.wrap(map);
+}
+
+export async function handleClientExamples(
+  context: TCGCContext,
+  client: SdkClientType<SdkServiceOperation>
+): Promise<[void, readonly Diagnostic[]]> {
+  const diagnostics = createDiagnosticCollector();
+
+  const examples = diagnostics.pipe(await loadExamples(context, client));
+  const methodQueue = [...client.methods];
```

Comment:
Be careful on the `client.methods` and its operationId.

When client.tsp defines the client or operation group, the operationId could be the `<interface>_<op>` in client.tsp, instead of routes.tsp.

Just make sure we've tested such case.